src/clr-angular/data/datagrid/datagrid-column-reorder-droppable.ts

```diff
@@ -4,12 +4,118 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component } from '@angular/core';
+import { Component, Input, Renderer2 } from '@angular/core';
+import { TableSizeService } from './providers/table-size.service';
+import { ColumnOrderModelService } from './providers/column-order-model.service';
+import { DomAdapter } from '../../utils/dom-adapter/dom-adapter';
+
+export const enum ColumnHeaderSides {
+  Left,
+  Right,
+}
 
 @Component({
   selector: 'clr-dg-column-reorder-droppable',
   template: `
-        <div class="datagrid-column-reorder-droppable"></div>
-    `,
+    <div class="datagrid-column-reorder-droppable" clrDroppable [clrGroup]="columnOrderDropKey"
+         (clrDragStart)="setDropTolerance($event)"
+         (clrDragEnter)="showHighlight(dropLine)"
+         (clrDragLeave)="hideHighlight(dropLine)"
+         (clrDrop)="updateOrder($event, dropLine)" [clrDropTolerance]="dropTolerance">
+      <div class="datagrid-column-drop-line" #dropLine></div>
+    </div>
+  `,
 })
-export class ClrDatagridColumnReorderDroppable {}
+export class ClrDatagridColumnReorderDroppable {
+  constructor(
+    private tableSizeService: TableSizeService,
+    private columnOrderModel: ColumnOrderModelService,
+    private renderer: Renderer2,
+    private domAdapter: DomAdapter
+  ) {}
+
+  public get columnOrderDropKey(): string {
+    return this.columnOrderModel.columnGroupId;
+  }
+
+  public dropTolerance: any;
+
+  // Each column headers will have this clr-dg-column-reorder-droppable component on each of its sides.
+  // We need a way to distinguish which side of the header that this droppable is on.
+
+  @Input('side') side: ColumnHeaderSides;
```

Comment:
This is covered in the tests. But this had to be tested indirectly because assigning a value alone to this property doesn't make any direct changes to the component itself as the value is used to calculate the drop tolerance.